pkg/framework/aries/framework.go

```diff
@@ -27,6 +27,8 @@ type Aries struct {
 	didResolver         DIDResolver
 	storeProvider       storage.Provider
 	protocolSvcCreators []api.ProtocolSvcCreator
+	ctx                 *context.Provider
+	services            []dispatcher.Service
```

Comment:
Seems odd to have both `dispatcher.Service` and `api.ProtocolSvcCreator` (the latter provide the former).

Reply:
`protocolSvcCreators` are passed a functional options and `services` are derived from it. We need this while creating a new context as we pass `services` to `context.WithProtocolServices()`. [Previously](https://github.com/hyperledger/aries-framework-go/pull/201/files#diff-9988f5820bf05228e5657f2842fa071eL23) these 2 were there in `context.Provider` and now moved to `Aries`. Do you have anything specific in mind ? I can change the code.

Reply:
It seems like we no longer need `api.ProtocolSvcCreator`, do we?

Reply:
we need it for the [option](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/framework/aries/framework.go#L83). And we can create `services` once we have ctx.

Reply:
@troyronda @rolsonquadras I'm not sure what direction to head in this case. Yes - context needs to be available in order to instantiate the protocol services. However, the user does not (currently) have access to a "default" context that they can use to instantiate their own protocol services and pass in to `framework.New()`. Perhaps we need to expose a `defaults.Context()` somewhere?

Reply:
@llorllale @troyronda this PR is just doing code refactor. The comments above looks more like a bigger design discussion. Can we this discussion moved to a new issue and merge this PR (if they are no blocking issues in this PR) ?

Reply:
It’s okay to me. What is the issue number?

Reply:
@troyronda @llorllale created #212